runtime/lua/vim/lsp.lua

```diff
@@ -798,11 +800,27 @@ function lsp.start_client(config)
     end
   end
 
+  local handles
+  -- Start the language server
+  if config.cmd then
```

Comment:
this should probably switch on some `mode` key to config, you can start a server *and* connect to it via TCP

Reply:
> this should probably switch on some `mode` key to config, you can start a server _and_ connect to it via TCP

like there should also be a named pipe/ unix socket option as well, as some servers (`powershell_es` for example) use named pipes to communicate

Reply:
Sure, named pipe support could be added but is out of scope of this PR which adds TCP support.